### ISSUE TITLE
Install the Collector

### ISSUE THREAD

**Issue by jingweiz2017** [open-telemetry/opentelemetry.io#3749]
"kubectl apply -f https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/main/examples/k8s/otel-config.yaml"
The command above will refer to the latest version of otel-config.yaml. There is a chance for its corresponding image has not been uploaded onto docker.io/otel yet. It will lead to a image pull backoff.

**Comment by svrnm** [comment]
@open-telemetry/collector-approvers PTAL!

**Comment by mx-psi** [comment]
> @mx-psi what's the approximate time window during a release where this may be problematic? Maybe adding a comment to the YML file can be helpful to debug **if** people run into this.

That would be from the prep release PR on https://github.com/open-telemetry/opentelemetry-collector getting merged up until the Github Action that releases the Docker image on https://github.com/open-telemetry/opentelemetry-collector-releases finishes. In between we have the https://github.com/open-telemetry/opentelemetry-collector-contrib release.

For example, in this release open-telemetry/opentelemetry-collector@40c7db876f048a576bb6833984dd8f8f1b88ef7d  happened at `Jan 10, 2024, 12:20 AM GMT+1`, the release was triggered at `Jan 10, 2024, 5:55 PM GMT+1` and finished at `Jan 10, 2024, 6:08 PM GMT+1` ([job](https://github.com/open-telemetry/opentelemetry-collector-releases/actions/runs/7478199352/job/20353016375)), or around ~18 hours from one change to the other. I would say this is the typical timeframe unless the contrib release gets blocked unexpectedly.

So, TL;DR, this is probably a problem every other week for the better part of one working day.

**Comment by svrnm** [comment]
That means in ~10% of the time this may happen (assuming this happens during a workday and there are 10 workdays in 2 weeks)? 

What if we replace the link with a versioned one and use the collector version update pull requests to bump that. The last one (https://github.com/open-telemetry/opentelemetry.io/pull/3753) happened at 6:17, so shortly after that job run.

```
https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/main/examples/k8s/otel-config.yaml
```

would become

```
https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/{{ params vers }}/examples/k8s/otel-config.yaml
```

**Comment by mx-psi** [comment]
> That means in ~10% of the time this may happen (assuming this happens during a workday and there are 10 workdays in 2 weeks)?

Yup, that's about it roughly.

> What if we replace the link with a versioned one and use the collector version update pull requests to bump that. The last one (https://github.com/open-telemetry/opentelemetry.io/pull/3753) happened at 6:17, so shortly after that job run. [...]

Makes sense to me to do that :+1: That would solve the issue, at least on the link that we have on the documentation.
